.github/workflows/samples_tests.yml

```diff
@@ -0,0 +1,134 @@
+name: Samples
+
+on:
+  workflow_call:
+    inputs:
+      runner:
+        description: 'Machine on which the tests would run'
+        type: string
+        required: true
+      image:
+        description: 'Docker image in which the tests would run'
+        type: string
+        required: false
+        default: null
+
+jobs:
+  Samples:
+    runs-on: ${{ inputs.runner }}
+    container:
+      image: ${{ inputs.image }}
+    env:
+      DEBIAN_FRONTEND: noninteractive # to prevent apt-get from waiting user input
+      INSTALL_DIR: ${{ github.workspace }}/install
+      INSTALL_TEST_DIR: ${{ github.workspace }}/install/tests
+      BUILD_DIR: ${{ github.workspace }}/build
+    steps:
+      - name: Download OpenVINO package (Linux)
+        if: runner.os == 'Linux'
+        uses: actions/download-artifact@v3
+        with:
+          name: openvino_package
+          path: ${{ env.INSTALL_DIR }}
+
+      - name: Download OpenVINO tests package (Linux)
+        if: runner.os == 'Linux'
+        uses: actions/download-artifact@v3
+        with:
+          name: openvino_tests
+          path: ${{ env.INSTALL_TEST_DIR }}
+
+      - name: Download OpenVINO package (mac)
+        if: runner.os == 'macOS'
+        uses: actions/download-artifact@v3
+        with:
+          name: openvino_package_${{ runner.arch }}
+          path: ${{ env.INSTALL_DIR }}
+
+      - name: Download OpenVINO tests package (mac)
+        if: runner.os == 'macOS'
+        uses: actions/download-artifact@v3
```

Comment:
why do we need conditional artifacts download?

Reply:
As the Mac workflow had both arm64 and x86 machines via a matrix - there is a need to distinguish between artefact -> it is done via appending the runner arch to the artefact name; the Linux workflow does not have that so there is a mismatch for artefact names between these workflows (`openvino_package` in Linux, `openvino_package_${{ runner.arch }}` in mac).
I've separated the Mac workflow into two as it would be for the Linux workflows, I'll see how it goes when I extract the tests (maybe builds too) into reusable workflows.

Reply:
separate workflows look better